src/Exception/RequestException.php

```diff
@@ -79,20 +79,25 @@ public static function create(
 
         $level = floor($response->getStatusCode() / 100);
         if ($level == '4') {
-            $label = 'Client error response';
+            $label = 'Client Error';
             $className = __NAMESPACE__ . '\\ClientException';
         } elseif ($level == '5') {
-            $label = 'Server error response';
+            $label = 'Server Error';
             $className = __NAMESPACE__ . '\\ServerException';
         } else {
-            $label = 'Unsuccessful response';
+            $label = 'Unsuccessful Request';
             $className = __CLASS__;
         }
 
-        $message = $label . ' [url] ' . $request->getUri()
-            . ' [http method] ' . $request->getMethod()
-            . ' [status code] ' . $response->getStatusCode()
-            . ' [reason phrase] ' . $response->getReasonPhrase();
+        // Server Error: `GET /` resulted in a `404 Not Found` response:
+        // <html> ... (truncated)
+        $message = sprintf(
+            "%s: `%s` resulted in a `%s` response:\n%s",
+            $label,
+            $request->getMethod() . ' ' . $request->getUri(),
+            $response->getStatusCode() . ' ' . $response->getReasonPhrase(),
+            substr((string) $response->getBody(), 0, 80)
```

Comment:
Sure, having a limited snippet of the response body could work, but there are some caveats we should consider:

1. We should use the `read` method of the response body instead of casting it to a string to ensure that we aren't loading more data into memory than needed.
2. We should only include the response body in the message if the body is seekable. If it is, and we read from the response, we should rewind the stream. This allows users to work with the stream in error handling.
3. This is maybe not necessary, but it could possibly be an issue for weirder APIs. Should we make sure that the response is printable? Maybe grab the first 80 chars and only include the response body if `ctype_print()` returns true.